Code/Framework/AzCore/AzCore/Debug/Trace.h

```diff
@@ -39,14 +37,117 @@ namespace AZ
+
+            /// Indicates if trace logging functions are enabled based on compile mode and cvar logging level
+            O3DEKERNEL_API bool IsTraceLoggingEnabledForLevel(LogLevel level)
+            {
+                return m_logLevel >= level;
+            }
+            O3DEKERNEL_API void SetLogLevel(LogLevel newLevel)
+            {
+                m_logLevel = newLevel;
+            }
 
+            O3DEKERNEL_API bool GetAlwaysPrintCallstack() const
+            {
+                return m_printCallstack;
+            }
+            O3DEKERNEL_API void SetAlwaysPrintCallstack(bool enable)
+            {
+                m_printCallstack = enable;
+            }
+
+            O3DEKERNEL_API virtual void Assert(const char* fileName, int line, const char* funcName, const char* format, ...)
+            {
+                char message[s_maxMessageLength];
+                va_list mark;
+                va_start(mark, format);
+                azvsnprintf(message, s_maxMessageLength, format, mark);
+                va_end(mark);
+                fprintf(stderr, "Assert: %s:%d (%s): %s\n", fileName, line, funcName, message);
+            }
+            O3DEKERNEL_API virtual void Error(const char* fileName, int line, const char* funcName, const char* window, const char* format, ...)
+            {
+                char message[s_maxMessageLength];
+                va_list mark;
+                va_start(mark, format);
+                azvsnprintf(message, s_maxMessageLength, format, mark);
+                va_end(mark);
+                fprintf(stderr, "Error: %s:%d (%s): %s: %s\n", fileName, line, funcName, window, message);
+            }
+            O3DEKERNEL_API virtual void Warning(const char* fileName, int line, const char* funcName, const char* window, const char* format, ...)
+            {
+                char message[s_maxMessageLength];
+                va_list mark;
+                va_start(mark, format);
+                azvsnprintf(message, s_maxMessageLength, format, mark);
+                va_end(mark);
+                fprintf(stderr, "Warning: %s:%d (%s): %s: %s\n", fileName, line, funcName, window, message);
+            }
+            O3DEKERNEL_API virtual void Printf(const char* window, const char* format, ...)
+            {
+                char message[s_maxMessageLength];
+                va_list mark;
+                va_start(mark, format);
+                azvsnprintf(message, s_maxMessageLength, format, mark);
+                va_end(mark);
+                fprintf(stdout, "%s: %s\n", window, message);
+            }
+            O3DEKERNEL_API virtual void Output(const char* window, const char* message)
+            {
+                fprintf(stdout, "%s: %s\n", window, message);
+            }
+            O3DEKERNEL_API virtual void RawOutput(const char* window, const char* message)
+            {
+                fprintf(stdout, "%s: %s\n", window, message);
```

Comment:
Raw Output should be `fwrite` to avoid any truncation of output to do internal printf buffers